### ISSUE TITLE
Issue Building iOS App to registered test device via XCode

### ISSUE THREAD

**Issue by BrandenSoropia** [AppsFlyerSDK/appsflyer-react-native-plugin#312]
# Report

## Plugin Version

`6.3.20`

## On what Platform are you having the issue?

iOS

## What did you do?

Try to build React Native (`0.63.3`) app to iOS device via XCode 12.4 and 12.5.1
1. `npx react-native init AwesomeProject --version 0.63.3`
2. `yarn add react-native-appsflyer`
3. `cd ios && pod install`
4. Open project in XCode via `AwesomeProject.xcworkspace`
5. Create your signing certificate/profile by:
    - In Project Navigator, click "ReactNativeAppsflyer".
    - Click the "Signing and Capabilities" tab.
    - Set "Team" to yourself (your Apple developer account)
    - Connect your iOS device to your computer. Select it as the target scheme to build to, by selecting your device in the drop down directly right of the play button in the top navbar.
    - It will automatically generate your certificate/profile now!
6. Select my connected iOS device in scheme. Press Play to build to it.

## What did you expect to happen?

App to build on test device.

## What happened instead?

XCode 12.4 and 12.5.1 reporting following error:
```In /Users/brandensoropia/Library/Developer/Xcode/DerivedData/ReactNativeAppsflyer-aqckwkdpnacpiwfgqrmtxiosvotv/Build/Products/Debug-iphoneos/AppsFlyerLib.framework/AppsFlyerLib(AFSDKDevice.o), building for iOS, but linking in object file built for Mac Catalyst, file '/Users/brandensoropia/Library/Developer/Xcode/DerivedData/ReactNativeAppsflyer-aqckwkdpnacpiwfgqrmtxiosvotv/Build/Products/Debug-iphoneos/AppsFlyerLib.framework/AppsFlyerLib' for architecture arm64```

## Please provide any other relevant information.

Full and fast setup demo here, with test results: https://github.com/BrandenSoropia/react-native-appsflyer

**Attempted Fixes**
I've tried the following fixes in hopes of getting around this, but no luck:
- Setting our app in Xcode’s Build Settings > “Build Active Architecture Only” to all “Yes”. 
Source: Xcode 12 beta 4 - error compiling for Simulator - https://developer.apple.com/forums/thread/656509?answerId=626387022#626387022

- Setting “Excluded Archs” to “arm64” in all build schemes. Source: https://stackoverflow.com/questions/63607158/xcode-12-building-for-ios-simulator-but-linking-in-object-file-built-for-ios

**Comment by pazlavi** [comment]
Hi @BrandenSoropia, 
Thank you for reaching out to us. 

Since V6.3.0, we are using XCFramework. To solve the issue, please update your Cocoapods version to the latest.

**Comment by BrandenSoropia** [comment]
Thanks for that! I did that and instead of just running `pod install` post `yarn add react-native-appsflyer`, I instead used `pod install --no-repo-update` to maintain my previously installed pods, and just install the new ones instead. Now it works perfectly, thank you!

So full run down, in my repo I did:
1. ` sudo gem install cocoapods` (install latest CocoaPods)
2. `yarn add react-native-appsflyer`
3. `cd ios && pod install --no-repo-update` (only install AppsFlyer pods, do not touch/update anything else)

Now both simulator and Xcode builds work!
